src/Interpreters/Context.cpp

```diff
@@ -1561,6 +1561,14 @@ void Context::setCurrentRolesDefault()
     setCurrentRolesImpl(user->granted_roles.findGranted(user->default_roles), /* throw_if_not_granted= */ false, /* skip_if_not_granted= */ false, user);
 }
 
+void Context::switchImpersonateUser(const RolesOrUsersSet & users )
```

Comment:
It seems this function is used only once and it's very simple. Let's move it to `InterpreterExecuteAs.cpp` perhaps?